apis/apps/v1/componentdefinition_types.go

```diff
@@ -1243,6 +1243,13 @@ type ComponentAvailable struct {
 	// +optional
 	WithPhases *string `json:"withPhases,omitempty"`
 
+	// Specifies the role that the component will go through to be considered available.
+	//
+	// This field is immutable once set.
+	//
+	// +optional
+	WithRole *string `json:"withRole,omitempty"`
```

Comment:
The current implementation treats a component as available, as long as any replica's role matches `withRole`. Can it be extended to support the semantics of "exactly one replica"?